legacy/routes/complex.js

```diff
@@ -10,6 +10,9 @@ var complex = function (coverage) {
    */
   router.put("/basic/:scenario", function (req, res, next) {
     if (req.params.scenario === "valid") {
+      if (req.query["api-version"] !== "2016-02-29") {
```

Comment:
is this scenario really the right one for this. I think if we want to do this test we should have a dedicated endpoint with a meaningful description of what the expected behavior is. Here it will just look like something added by mistake

Reply:
Hi @timotheeguerin according to swagger definition: https://github.com/Azure/autorest.testserver/blob/1d0ec0cf2e0944346d3f8cf93980f392b82eb230/swagger/body-complex.json#L1759-L1767

`api-version` is required and only has one value, so the check is necessary. If it exists before, python could find the [bug](https://github.com/Azure/autorest.python/issues/1267) earlier.

Reply:
hhm, I guess yeah the spec does say that but it does feel like a terrible scenario not doing what its supposed to(Looking about body content). In the best world we should have a new `client-parameters.json` swagger spec and add the scenarios there

Reply:
that makes sense. I will close the PR.
Thanks for the review.